platform-server-kernel/src/main/java/uno/cod/platform/server/core/service/InvitationService.java

```diff
@@ -0,0 +1,103 @@
+        this.userService = userService;
+        this.organizationService = organizationService;
+        this.challengeService = challengeService;
+        this.environment = environment;
+    }
+
+    public void invite(InvitationDto dto) {
+        OrganizationShowDto organization = organizationService.findById(dto.getOrganizationId());
+        if (organization == null) {
+            throw new IllegalArgumentException("invalid organization");
+        }
+        ChallengeShowDto challenge = challengeService.findById(dto.getChallengeId());
+        if (challenge == null) {
+            throw new IllegalArgumentException("invalid challenge");
+        }
+        UserShowDto user = userService.findByEmail(dto.getEmail());
+        if (user == null) {
+            UserCreateDto userCreateDto = new UserCreateDto();
+            userCreateDto.setEmail(dto.getEmail());
+            userCreateDto.setNick(dto.getEmail());
+            userCreateDto.setPassword("random_password");
+            userService.createFromDto(userCreateDto);
+        }
+        user = userService.findByEmail(dto.getEmail());
+        String mailMessage = getChallengeStartMailMessage(user, organization, challenge);
+
+        if (Arrays.asList(this.environment.getActiveProfiles()).contains("development")) {
+            log.log(Level.ERROR, mailMessage);
+        } else {
+            // TODO - send mail
+            MimeMessage message = null;
+            try {
+                Transport.send(message);
+            } catch (MessagingException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private String getChallengeStartMailMessage(UserShowDto user, OrganizationShowDto organization, ChallengeShowDto challenge) {
+        return "Hi " + user.getUsername() + ",\n" +
+                "\n" +
+                "You are invited to test a preview version of Coduno for " + organization.getName() + ".\n" +
+                "\n" +
+                "This means you are asked to do a coding assessment that will require a range of skills\n" +
+                "in connection with software engineering. You might want to brush up your skills before\n" +
+                "taking up the challenge.\n" +
+                "\n" +
+                "Once you feel confident and have at least an hour of spare time, you can access the\n" +
+                "challenge via:\n" +
+                "\n\t" +
+                getAppUrl() + "#!/token/"+ getToken(challenge.getId()) +
+                "\n\n" +
+                "As soon as you point your browser there, the challenge is on!\n" +
+                "\n" +
+                "Further instructions are provided there.\n" +
+                "\n" +
+                "If you have trouble, contact the Coduno team at +43 680 3031228.\n" +
+                "\n" +
+                "Good luck!\n" +
+                "Coduno";
+    }
+
+    private String getAppUrl() {
```

Comment:
injected to the mail templates via ${url} can be changed in the application*.yml